File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int majorElement = 0, majorityValue = 0;
        for (int i = 0; i < nums.length; i++) {
            if (majorityValue == 0) {
                majorElement = nums[i];
            }
            if (nums[i] == majorElement) {
                majorityValue++;
            } else {
                majorityValue--;
            }
        }
        return majorElement;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        ans = 1
        max_product = 1
        min_product = 1

        for num in nums:
            max_product *= num
            min_product *= num

            max_product = max(max_product, min_product, num)
            min_product = min(max_product, min_product, num)
            ans = max(ans, max_product)
        return ans
        